internal/pkg/api/handleEnroll_test.go

```diff
@@ -90,5 +95,182 @@ func TestEnroll(t *testing.T) {
 	if resp.Action != "created" {
 		t.Fatal("enroll failed")
 	}
+}
+
+func TestEnrollerT_fetchStaticTokenPolicy(t *testing.T) {
+	cacheBuilder := func(policies ...model.Policy) func() cache.Cache {
+		return func() cache.Cache {
+			c, _ := cache.New(config.Cache{
+				NumCounters:  100,
+				MaxCost:      100000,
+				PolicyKeyTTL: time.Hour,
+			})
+			for _, p := range policies {
+				c.SetPolicy(p)
+			}
+			return c
+		}
+	}
+	bulkerBuilder := func(policies ...model.Policy) func() bulk.Bulk {
+		return func() bulk.Bulk {
+			bulker := ftesting.NewMockBulk()
 
+			hits := []es.HitT{}
+			for _, p := range policies {
+				b, _ := json.Marshal(p)
+				hits = append(hits, es.HitT{
+					Source: b,
+				})
+			}
+			res := &es.ResultT{
+				HitsT: es.HitsT{},
+				Aggregations: map[string]es.Aggregation{
+					"policy_id": {
+						Buckets: []es.Bucket{
+							{
+								Aggregations: map[string]es.HitsT{
+									"revision_idx": {
+										Hits: hits,
+									},
+								},
+							},
+						},
+					},
+				},
+			}
+			bulker.On("Search", mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(res, nil)
+			return bulker
+		}
+	}
+	type fields struct {
+		staticPolicyTokens config.StaticPolicyTokens
+		cache              func() cache.Cache
+		bulker             func() bulk.Bulk
+	}
+	type args struct {
+		enrollmentAPIKey *apikey.APIKey
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    *model.EnrollmentAPIKey
+		wantErr bool
+	}{
+		{
+			name: "disabled",
+			fields: fields{
+				staticPolicyTokens: config.StaticPolicyTokens{
+					Enabled: false,
+				},
+				cache:  cacheBuilder(),
+				bulker: bulkerBuilder(),
+			},
+			args: args{
+				enrollmentAPIKey: &apikey.APIKey{},
+			},
+			want:    nil,
+			wantErr: false,
+		},
+		// flaky test
```

Comment:
For some reason, this is a flaky one.
We see logs such as
```bash
{"level":"trace","ok":true,"id":"dummy-policy","cost":16,"ttl":3600000,"time":"2023-06-20T12:19:04+03:00","message":"Policy cache SET"}
{"level":"trace","id":"dummy-policy","time":"2023-06-20T12:19:04+03:00","message":"Policy cache MISS"}
```
Where the cache was set, but it was missed later.